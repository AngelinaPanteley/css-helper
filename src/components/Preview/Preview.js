import React, { PureComponent } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

class Preview extends PureComponent {
  static propTypes = {
    template: PropTypes.string.isRequired,
    previewClass: PropTypes.string.isRequired,
    styles: PropTypes.object.isRequired,
  }

  componentDidMount() {
    this.setStyles();
  }

  componentDidUpdate() {
    this.setStyles();
  }

  createMarkup = () => {
    return { __html: this.props.template };
  }

  setStyles = () => {
    const previewClass = this.props.previewClass;
    const element = ReactDOM.findDOMNode(this.refs.wrapper)
      .getElementsByClassName(previewClass).item(previewClass);

    for (let styleKey in this.props.styles) {
      element.style[styleKey] = this.props.styles[styleKey];
    }
  }

  render() {
    return (
      <div ref="wrapper"
        dangerouslySetInnerHTML={this.createMarkup()} >
      </div>
    );
  }
}

export default Preview;