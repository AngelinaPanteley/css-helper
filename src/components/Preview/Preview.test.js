"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';

import Preview from './Preview.js';

configure({ adapter: new Adapter() });

describe('>>>Preview', () => {

  //SNAPSHOTS

  const props = {
    template: "<div class='template'></div>",
    previewClass: 'template',
    styles: {
      border: '1px solid gray',
    },
  };

  const previewComponent = <Preview {...props} />;

  const component = renderer.create(previewComponent);

  it('renders preview correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
})