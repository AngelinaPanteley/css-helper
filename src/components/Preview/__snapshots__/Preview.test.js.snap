// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Preview renders preview correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div class='template'></div>",
    }
  }
/>
`;
